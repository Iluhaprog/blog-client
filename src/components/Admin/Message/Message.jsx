import React from 'react';
import {connect} from 'react-redux';
import {removeMessage} from '../../../store/message/messageActions';
import styled from 'styled-components';
import {Alert, Button, Container, Row} from 'react-bootstrap';
import {MessageTypes} from '../../../store/message/MessageTypes';
import PropTypes from 'prop-types';
import {XSquare} from 'react-bootstrap-icons';

const MessagesBox = styled.div`
  position: fixed;
  bottom: 0;
  right: 0;
  max-width: 250px;
  max-height: 85vh;
  width: 100%;
  overflow-y: scroll;
  background-color: transparent;
  border-radius: 5px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
`;

let Message = ({messages, messageLiveTime, removeMessage}) => {
  const messageVariants = {
    [MessageTypes.ERROR]: 'danger',
    [MessageTypes.SUCCESS]: 'success',
  };

  return (
    <MessagesBox>
      {
        messages.map((message) => {
          setTimeout(() => {
            removeMessage(message.id);
          }, messageLiveTime);
          return (
            <Alert
              key={message.id}
              variant={messageVariants[message.type]}
            >
              <Container>
                <Row className='justify-content-between align-items-center'>
                  {
                    message.data.isAxiosError ?
                      (message.data.response.data.message) :
                      'Success'
                  }
                  <Button
                    variant={messageVariants[message.type]}
                    onClick={() => removeMessage(message.id)}
                  >
                    <XSquare />
                  </Button>
                </Row>
              </Container>
            </Alert>
          );
        })
      }
    </MessagesBox>
  );
};

Message.propTypes = {
  messages: PropTypes.array,
  messageLiveTime: PropTypes.number,
  removeMessage: PropTypes.func,
};

const mapStateToProps = (state) => ({
  messages: state.message.messages,
  messageLiveTime: +state.message.messageLiveTime,
});

const mapDispatchToProps = (dispatch) => ({
  removeMessage: (id) => dispatch(removeMessage(id)),
});

Message = connect(mapStateToProps, mapDispatchToProps)(Message);

export {Message};
