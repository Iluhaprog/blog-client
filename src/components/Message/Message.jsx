import React from 'react';
import { MessagesBox } from '../boxes';
import { Error as ErrorMessage } from '../messages';
import { CSSTransition, TransitionGroup } from 'react-transition-group';

export default ({ errors = [], show = () => {} }) => (
    <MessagesBox>
        <TransitionGroup>
        {
            errors.map(error => {
                setTimeout(() => {
                    show(error.id);
                }, 10000);

                return (
                    <CSSTransition
                        key={error.id}
                        timeout={300}
                        classNames='error-box'
                    >
                        <ErrorMessage 
                            message={error.error.message}
                            onClose={() => show(error.id)}
                        />
                    </CSSTransition>
                )
            })
        }
        </TransitionGroup>
    </MessagesBox>
);