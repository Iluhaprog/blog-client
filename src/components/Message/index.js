import msg from './Message';
import { connect } from 'react-redux';
import { showError } from '../../actoins/error';

const mapStateToProps = state => ({
    errors: state.error.array.filter(error => !error.showed),
});

const mapDispatchToProps = dispatch => ({
    show: id => {
        dispatch(showError(id));
    }
});

const Message = connect(mapStateToProps, mapDispatchToProps)(msg);

export {
    Message,
};