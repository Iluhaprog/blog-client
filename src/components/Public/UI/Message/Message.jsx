import styled from 'styled-components';

const Message = styled.h3`
  width: 100%;
  padding-top: 40px;
  text-align: center;
  color: #7f7f7f;
`;

export {Message};
