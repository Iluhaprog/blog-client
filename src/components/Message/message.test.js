import React from 'react';
import { render } from '@testing-library/react';
import Message from './Message';

test('Test view of message box without children', () => {
    const { asFragment } = render(<Message />);
    expect(asFragment()).toMatchSnapshot();
});


test('Test view of message box with children', () => {
    const er = [{
        id: 1,
        showed: false,
        error: { message: 'test' },
    }];
    const { asFragment } = render(<Message errors={er}/>);
    expect(asFragment()).toMatchSnapshot();
})
