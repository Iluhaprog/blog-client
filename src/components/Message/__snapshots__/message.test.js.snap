// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view of message box with children 1`] = `
<DocumentFragment>
  <section
    class="messages-box"
  >
    <div>
      <div
        class="error-box"
      >
        <div
          class="row   "
        >
          <p
            class="error-box__message"
          >
            test
          </p>
          <button
            class="button default-button close-button"
          >
            X
          </button>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Test view of message box without children 1`] = `
<DocumentFragment>
  <section
    class="messages-box"
  >
    <div />
  </section>
</DocumentFragment>
`;
